modules/codegen/test/src/smithy4s/codegen/internals/DefaultRenderModeSpec.scala

```diff
@@ -151,4 +151,133 @@ final class DefaultRenderModeSpec extends munit.FunSuite {
     TestUtils.runTest(smithy, scalaCode)
   }
 
+  test("allow for self referencing trait - structure") {
```

Comment:
**issue:** I am sure that this is not the right place for these kind of tests, but I wasn't sure where to put them. Also, I don't need to check the generated code, all I need is to assert on the IR.

Reply:
Instead of doing a rendering test, you can do a smoke test in the bootstrapped module, adding a smithy file to the sample specs and exercising the code in the tests to verify that the schema doesn't blow up during its initialisation